include/idf/EthernetDevice.hh

```diff
@@ -0,0 +1,164 @@
+/*
+PURPOSE:
+LIBRARY DEPENDENCIES: (
+(idf/EthernetDevice.cpp)
+)
+*/
+
+/**
+ * @trick_parse{everything}
+ * @trick_link_dependency{idf/EthernetDevice.cpp}
+ */
+
+#ifndef ETHERNET_DEVICE_HH
+#define ETHERNET_DEVICE_HH
+
+#include "idf/InputDevice.hh"
+
+#include <string>
+#include <netinet/in.h>
+
+namespace idf {
+
+/**
+ * @brief abstract base class for all ethernet input devices. Subclasses should usually
+ * represet a specific physical device and implement logic to fulfill the contract of
+ * open(). They should also usually derive from the appropriate class from the %Input
+ * Abstration Layer, as well as be concrete.
+ * 
+ * @author  Philip Kunz
+ */
+class EthernetDevice : public InputDevice {
+
+    public:
+
+    /**
+     * @brief constructs a new instance whose open() will open the UDP port at
+     * @a ipAddress: @a port
+     * 
+     * @param ipAddress the ipaddress of this device
+     * @param port the port number of this device
+     */
+    EthernetDevice(const std::string& name, unsigned packetLength);
+
+    virtual ~EthernetDevice() {
+        close();
+    };
+    
+    virtual void open();
```

Comment:
if you can use c++11, these virtual methods can all have the override keyword for methods inherited from InputDevice.